diff --git a/src/Controller/SecurityController.php b/src/Controller/SecurityController.php
index 581f27b..daf9106 100644
--- a/src/Controller/SecurityController.php
+++ b/src/Controller/SecurityController.php
@@ -4,6 +4,7 @@ namespace App\Controller;
 
 use Doctrine\ORM\EntityManagerInterface;
 use Scheb\TwoFactorBundle\Security\TwoFactor\Provider\Totp\TotpAuthenticatorInterface;
+use Scheb\TwoFactorBundle\Security\TwoFactor\QrCode\QrCodeGenerator;
 use Sensio\Bundle\FrameworkExtraBundle\Configuration\IsGranted;
 use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
 use Symfony\Component\HttpFoundation\Response;
@@ -51,8 +52,10 @@ class SecurityController extends AbstractController
      * @Route("/authentication/2fa/qr-code", name="app_qr_code")
      * @IsGranted("IS_AUTHENTICATED_FULLY")
      */
-    public function displayGoogleAuthenticatorQrCode()
+    public function displayGoogleAuthenticatorQrCode(QrCodeGenerator $qrCodeGenerator)
     {
-        die('TODO!');
+        $qrCode = $qrCodeGenerator->getTotpQrCode($this->getUser());
+
+        return new Response($qrCode->writeString(), 200, ['Content-Type' => 'image/png']);
     }
 }
