diff --git a/src/Controller/SecurityController.php b/src/Controller/SecurityController.php
index 67fbc87..dbf36fe 100644
--- a/src/Controller/SecurityController.php
+++ b/src/Controller/SecurityController.php
@@ -38,7 +38,6 @@ class SecurityController extends AbstractController
             $entityManager->flush();
         }
 
-        dd($this->getUser());
         return $this->render('security/enable2fa.html.twig');
     }
 
