diff --git a/src/Entity/User.php b/src/Entity/User.php
index 51b30d2..47b2ea7 100644
--- a/src/Entity/User.php
+++ b/src/Entity/User.php
@@ -4,6 +4,7 @@ namespace App\Entity;
 
 use App\Repository\UserRepository;
 use Doctrine\ORM\Mapping as ORM;
+use Scheb\TwoFactorBundle\Model\Totp\TotpConfiguration;
 use Scheb\TwoFactorBundle\Model\Totp\TotpConfigurationInterface;
 use Scheb\TwoFactorBundle\Model\Totp\TwoFactorInterface;
 use Symfony\Component\Security\Core\User\PasswordAuthenticatedUserInterface;
@@ -176,16 +177,16 @@ class User implements UserInterface, PasswordAuthenticatedUserInterface, TwoFact
 
     public function isTotpAuthenticationEnabled(): bool
     {
-        // TODO: Implement isTotpAuthenticationEnabled() method.
+        return $this->totpSecret ? true : false;
     }
 
     public function getTotpAuthenticationUsername(): string
     {
-        // TODO: Implement getTotpAuthenticationUsername() method.
+        return $this->getUserIdentifier();
     }
 
     public function getTotpAuthenticationConfiguration(): ?TotpConfigurationInterface
     {
-        // TODO: Implement getTotpAuthenticationConfiguration() method.
+        return new TotpConfiguration($this->totpSecret, TotpConfiguration::ALGORITHM_SHA1, 30, 6);
     }
 }
