diff --git a/templates/security/enable2fa.html.twig b/templates/security/enable2fa.html.twig
index 168a0f4..7eca7d7 100644
--- a/templates/security/enable2fa.html.twig
+++ b/templates/security/enable2fa.html.twig
@@ -8,7 +8,7 @@
         <div class="login-form bg-light mt-4 p-4">
             <h1 class="h3 mb-3 font-weight-normal">Use Authy or Google Authenticator to Scan the QR Code</h1>
 
-            <img src="TODO" alt="2fa QR Code">
+            <img src="{{ path('app_qr_code') }}" alt="2fa QR Code">
         </div>
     </div>
 </div>
