diff --git a/src/Security/QueryParamsAuthenticator.php b/src/Security/QueryParamsAuthenticator.php
index f952fa5..5d4924a 100644
--- a/src/Security/QueryParamsAuthenticator.php
+++ b/src/Security/QueryParamsAuthenticator.php
@@ -18,7 +18,7 @@ class QueryParamsAuthenticator extends AbstractAuthenticator
             && $request->query->has('password');
     }
 
-    public function authenticate(Request $request)
+    public function authenticate(Request $request): Passport
     {
         dd('authenticate!');
     }
