diff --git a/src/Controller/ApiController.php b/src/Controller/ApiController.php
index b4c6f52..b2152b2 100644
--- a/src/Controller/ApiController.php
+++ b/src/Controller/ApiController.php
@@ -14,4 +14,12 @@ class ApiController extends AbstractController
     {
         return $this->json($this->getUser());
     }
+
+    /**
+     * @Route("/api/login", name="api_login", methods={"POST"})
+     */
+    public function apiLogin()
+    {
+        throw new \Exception('This will never be executed: intercepted by JsonLoginAuthenticator.');
+    }
 }
diff --git a/src/Security/JsonLoginAuthenticator.php b/src/Security/JsonLoginAuthenticator.php
index 1848606..1d58538 100644
--- a/src/Security/JsonLoginAuthenticator.php
+++ b/src/Security/JsonLoginAuthenticator.php
@@ -25,8 +25,7 @@ class JsonLoginAuthenticator extends AbstractAuthenticator
 
     public function supports(Request $request): ?bool
     {
-        return $request->query->has('email')
-            && $request->query->has('password');
+        return $request->getPathInfo() === '/api/login';
     }
 
     public function authenticate(Request $request): Passport
