diff --git a/src/Security/JwtAuthenticator.php b/src/Security/JwtAuthenticator.php
index ce3e5d3..8ab7e56 100644
--- a/src/Security/JwtAuthenticator.php
+++ b/src/Security/JwtAuthenticator.php
@@ -57,7 +57,7 @@ class JwtAuthenticator extends AbstractAuthenticator
 
     public function onAuthenticationSuccess(Request $request, TokenInterface $token, string $firewallName): ?Response
     {
-        // TODO: Implement onAuthenticationSuccess() method.
+        return null;
     }
 
     public function onAuthenticationFailure(Request $request, AuthenticationException $exception): ?Response
