diff --git a/src/Security/JwtAuthenticator.php b/src/Security/JwtAuthenticator.php
index e0522da..ed2a956 100644
--- a/src/Security/JwtAuthenticator.php
+++ b/src/Security/JwtAuthenticator.php
@@ -13,12 +13,13 @@ class JwtAuthenticator extends AbstractAuthenticator
 {
     public function supports(Request $request): ?bool
     {
-        // TODO: Implement supports() method.
+        return $request->headers->has('Authorization');
     }
 
     public function authenticate(Request $request): Passport
     {
-        // TODO: Implement authenticate() method.
+        $authorization = $request->headers->get('Authorization');
+        dd($authorization);
     }
 
     public function onAuthenticationSuccess(Request $request, TokenInterface $token, string $firewallName): ?Response
diff --git a/tests/ApiFunctionalTest.php b/tests/ApiFunctionalTest.php
index 8b0bd73..d986dbd 100644
--- a/tests/ApiFunctionalTest.php
+++ b/tests/ApiFunctionalTest.php
@@ -24,7 +24,6 @@ class ApiFunctionalTest extends KernelTestCase
             ->assertJson()
         ;
         $data = json_decode($browser->response()->body(), true);
-        dd($data);
 
         $this->browser()
             ->get('/api/me', [
