diff --git a/src/EventListener/LastLoggedInAtSubscriber.php b/src/EventListener/LastLoggedInAtSubscriber.php
index b269a82..e761224 100644
--- a/src/EventListener/LastLoggedInAtSubscriber.php
+++ b/src/EventListener/LastLoggedInAtSubscriber.php
@@ -4,6 +4,7 @@ namespace App\EventListener;
 
 use App\Entity\User;
 use Doctrine\ORM\EntityManagerInterface;
+use Scheb\TwoFactorBundle\Security\Authentication\Token\TwoFactorToken;
 use Symfony\Component\EventDispatcher\EventSubscriberInterface;
 use Symfony\Component\Security\Http\Event\LoginSuccessEvent;
 
@@ -18,6 +19,10 @@ class LastLoggedInAtSubscriber implements EventSubscriberInterface
 
     public function onLoginSuccess(LoginSuccessEvent $event)
     {
+        if ($event->getAuthenticatedToken() instanceof TwoFactorToken) {
+            return;
+        }
+
         $user = $event->getUser();
         if (!$user instanceof User) {
             throw new \Exception('Invalid user');
